### PR TITLE
sinkv2(cdc): correct source and sink closing order

### LINKED ISSUE #7762: CDC panic: Table sink not found when updating resolved ts
### What did you do?

Run test case: redo_apply_multiple_workload_node_restart_sync.
- create redo log changefeed
- Run workload
- inject allCaptureRestart, and allPDRestart

### What did you expect to see?

No panic

### What did you see instead?

[root@centos76_vm debugCDC]# kubectl --kubeconfig kubeconfig.yml -n cdc-testbed-tps-1323421-1-22 logs -p upstream-cdc-0
Error from server (NotFound): pods "upstream-cdc-0" not found
[root@centos76_vm debugCDC]# kubectl --kubeconfig kubeconfig.yml -n cdc-testbed-tps-1323421-1-22 logs -p upstream-ticdc-0
[WARN] TiCDC server data-dir is not set. Please use `cdc server --data-dir` to start the cdc server if possible.
{"level":"warn","ts":"2022-12-01T06:35:45.183Z","logger":"etcd-client","caller":"v3@v3.5.4/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000e3bc00/upstream-pd:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest balancer error: last connection error: connection error: desc = \"transport: Error while dialing dial tcp [10.233.125.174:2379](http://10.233.125.174:2379/): connect: connection refused\""}
{"level":"warn","ts":"2022-12-01T06:35:51.185Z","logger":"etcd-client","caller":"v3@v3.5.4/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000e3bc00/upstream-pd:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest balancer error: last connection error: connection error: desc = \"transport: Error while dialing dial tcp [10.233.125.174:2379](http://10.233.125.174:2379/): connect: connection refused\""}
{"level":"warn","ts":"2022-12-01T06:35:57.186Z","logger":"etcd-client","caller":"v3@v3.5.4/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000e3bc00/upstream-pd:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest balancer error: last connection error: connection error: desc = \"transport: Error while dialing dial tcp [10.233.125.174:2379](http://10.233.125.174:2379/): connect: connection refused\""}
{"level":"warn","ts":"2022-12-01T06:36:03.187Z","logger":"etcd-client","caller":"v3@v3.5.4/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000e3bc00/upstream-pd:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest balancer error: last connection error: connection error: desc = \"transport: Error while dialing dial tcp [10.233.125.174:2379](http://10.233.125.174:2379/): connect: connection refused\""}
{"level":"warn","ts":"2022-12-01T06:36:05.177Z","logger":"etcd-client","caller":"v3@v3.5.4/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000e3bc00/upstream-pd:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest balancer error: last connection error: connection error: desc = \"transport: Error while dialing dial tcp [10.233.125.174:2379](http://10.233.125.174:2379/): connect: connection refused\""}
{"level":"info","ts":"2022-12-01T06:36:05.177Z","logger":"etcd-client","caller":"v3@v3.5.4/client.go:210","msg":"Auto sync endpoints failed.","error":"context deadline exceeded"}
{"level":"warn","ts":"2022-12-01T06:36:09.188Z","logger":"etcd-client","caller":"v3@v3.5.4/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000e3bc00/upstream-pd:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest balancer error: last connection error: connection error: desc = \"transport: Error while dialing dial tcp [10.233.125.174:2379](http://10.233.125.174:2379/): connect: connection refused\""}
panic: add an exist table

goroutine 2754 [running]:
[go.uber.org/zap/zapcore.CheckWriteAction.OnWrite](http://go.uber.org/zap/zapcore.CheckWriteAction.OnWrite)(0x3?, 0x3?, {0x0?, 0x0?, 0xc000647000?})
        [go.uber.org/zap@v1.23.0/zapcore/entry.go:198](http://go.uber.org/zap@v1.23.0/zapcore/entry.go:198) +0x65
[go.uber.org/zap/zapcore.(*CheckedEntry).Write](http://go.uber.org/zap/zapcore.(*CheckedEntry).Write)(0xc000a94ea0, {0xc003e04cc0, 0x3, 0x3})
        [go.uber.org/zap@v1.23.0/zapcore/entry.go:264](http://go.uber.org/zap@v1.23.0/zapcore/entry.go:264) +0x3ec
[go.uber.org/zap.(*Logger).Panic](http://go.uber.org/zap.(*Logger).Panic)(0xc003e04cc0?, {0x381cc8e?, 0xc005a657c8?}, {0xc003e04cc0, 0x3, 0x3})
        [go.uber.org/zap@v1.23.0/logger.go:251](http://go.uber.org/zap@v1.23.0/logger.go:251) +0x59
[github.com/pingcap/log.Panic](http://github.com/pingcap/log.Panic)({0x381cc8e?, 0xc003b376b0?}, {0xc003e04cc0?, 0x7fea013165b8?, 0x3f95525b90c875ba?})
        [github.com/pingcap/log@v1.1.1-0.20221116035753-734d527bc87c/global.go:54](http://github.com/pingcap/log@v1.1.1-0.20221116035753-734d527bc87c/global.go:54) +0x8b
[github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/pebble.(*EventSorter).AddTable](http://github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/pebble.(*EventSorter).AddTable)(0xc0020ffe40, 0x10f5)
        [github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/pebble/event_sorter.go:102](http://github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/pebble/event_sorter.go:102) +0x2be
[github.com/pingcap/tiflow/cdc/processor/sourcemanager.(*SourceManager).AddTable](http://github.com/pingcap/tiflow/cdc/processor/sourcemanager.(*SourceManager).AddTable)(0xc006815810, {0x40fb030, 0xc003ce4d50}, 0x10f5, {0xc003ccb460, 0x4}, 0x6133193f1580005)
        [github.com/pingcap/tiflow/cdc/processor/sourcemanager/manager.go:70](http://github.com/pingcap/tiflow/cdc/processor/sourcemanager/manager.go:70) +0x18c
[github.com/pingcap/tiflow/cdc/processor.(*processor).AddTable](http://github.com/pingcap/tiflow/cdc/processor.(*processor).AddTable)(0xc000756c00, {0x7fe9d9a4f058?, 0xc003ce4d50}, 0x10f5, 0x6133193f1580005, 0x1)
        [github.com/pingcap/tiflow/cdc/processor/processor.go:217](http://github.com/pingcap/tiflow/cdc/processor/processor.go:217) +0x1a5f
[github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent.(*table).handleAddTableTask](http://github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent.(*table).handleAddTableTask)(0xc00523ae10, {0x7fe9d9a4f058, 0xc003ce4d50})
        [github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent/table.go:159](http://github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent/table.go:159) +0x442
[github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent.(*table).poll](http://github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent.(*table).poll)(0xc005a66fa0?, {0x7fe9d9a4f058?, 0xc003ce4d50?})
        [github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent/table.go:272](http://github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent/table.go:272) +0x46
[github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent.(*tableManager).poll](http://github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent.(*tableManager).poll)(0xc00666cd80, {0x7fe9d9a4f058, 0xc003ce4d50})
        [github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent/table.go:295](http://github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent/table.go:295) +0x109
[github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent.(*agent).Tick](http://github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent.(*agent).Tick)(0xc00797c000, {0x7fe9d9a4f058, 0xc003ce4d50})
        [github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent/agent.go:180](http://github.com/pingcap/tiflow/cdc/scheduler/internal/v3/agent/agent.go:180) +0x7e
[github.com/pingcap/tiflow/cdc/processor.(*processor).tick](http://github.com/pingcap/tiflow/cdc/processor.(*processor).tick)(0xc000756c00, {0x40fb030?, 0xc003ce4d50})
        [github.com/pingcap/tiflow/cdc/processor/processor.go:726](http://github.com/pingcap/tiflow/cdc/processor/processor.go:726) +0x1b8
[github.com/pingcap/tiflow/cdc/processor.(*processor).Tick](http://github.com/pingcap/tiflow/cdc/processor.(*processor).Tick)(0xc000756c00, {0x40fb030, 0xc003ce4d50})
        [github.com/pingcap/tiflow/cdc/processor/processor.go:643](http://github.com/pingcap/tiflow/cdc/processor/processor.go:643) +0x3a5
[github.com/pingcap/tiflow/cdc/processor.(*managerImpl).Tick](http://github.com/pingcap/tiflow/cdc/processor.(*managerImpl).Tick)(0xc0057d40c0, {0x7fe9d9a4dfc8?, 0xc007e8cf40}, {0x40c23a0?, 0xc00766e480})
        [github.com/pingcap/tiflow/cdc/processor/manager.go:128](http://github.com/pingcap/tiflow/cdc/processor/manager.go:128) +0x40c
[github.com/pingcap/tiflow/pkg/orchestrator.(*EtcdWorker).Run](http://github.com/pingcap/tiflow/pkg/orchestrator.(*EtcdWorker).Run)(0xc0076d4f20, {0x7fe9d9a4dfc8, 0xc007e8cf40}, 0xc0052c5680, 0x2faf080, {0x37fb19a, 0x9})
        [github.com/pingcap/tiflow/pkg/orchestrator/etcd_worker.go:290](http://github.com/pingcap/tiflow/pkg/orchestrator/etcd_worker.go:290) +0xfa6
[github.com/pingcap/tiflow/cdc/capture.(*captureImpl).runEtcdWorker](http://github.com/pingcap/tiflow/cdc/capture.(*captureImpl).runEtcdWorker)(0xc000b1ab40, {0x40fb0d0?, 0xc007e8cf40}, {0x7fe9d9a4e000, 0xc0057d40c0}, {0x40c23a0, 0xc00766e480}, 0xc00246de80?, {0x37fb19a, 0x9})
        [github.com/pingcap/tiflow/cdc/capture/capture.go:515](http://github.com/pingcap/tiflow/cdc/capture/capture.go:515) +0x185
[github.com/pingcap/tiflow/cdc/capture.(*captureImpl).run.func3()](http://github.com/pingcap/tiflow/cdc/capture.(*captureImpl).run.func3())
        [github.com/pingcap/tiflow/cdc/capture/capture.go:369](http://github.com/pingcap/tiflow/cdc/capture/capture.go:369) +0x299
[golang.org/x/sync/errgroup.(*Group).Go.func1()](http://golang.org/x/sync/errgroup.(*Group).Go.func1())
        [golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75](http://golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75) +0x64
created by [golang.org/x/sync/errgroup.(*Group).Go](http://golang.org/x/sync/errgroup.(*Group).Go)
        [golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72](http://golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72) +0xa5

====
[root@centos76_vm debugCDC]# kubectl --kubeconfig kubeconfig.yml -n cdc-testbed-tps-1323421-1-22 logs -p upstream-ticdc-2
[WARN] TiCDC server data-dir is not set. Please use `cdc server --data-dir` to start the cdc server if possible.
{"level":"warn","ts":"2022-12-01T06:35:46.831Z","logger":"etcd-client","caller":"v3@v3.5.4/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000d33500/upstream-pd:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest balancer error: last connection error: connection error: desc = \"transport: Error while dialing dial tcp [10.233.125.174:2379](http://10.233.125.174:2379/): connect: connection refused\""}
{"level":"warn","ts":"2022-12-01T06:35:52.832Z","logger":"etcd-client","caller":"v3@v3.5.4/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000d33500/upstream-pd:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest balancer error: last connection error: connection error: desc = \"transport: Error while dialing dial tcp [10.233.125.174:2379](http://10.233.125.174:2379/): connect: connection refused\""}
{"level":"warn","ts":"2022-12-01T06:35:58.833Z","logger":"etcd-client","caller":"v3@v3.5.4/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000d33500/upstream-pd:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest balancer error: last connection error: connection error: desc = \"transport: Error while dialing dial tcp [10.233.125.174:2379](http://10.233.125.174:2379/): connect: connection refused\""}
{"level":"warn","ts":"2022-12-01T06:36:04.835Z","logger":"etcd-client","caller":"v3@v3.5.4/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000d33500/upstream-pd:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest balancer error: last connection error: connection error: desc = \"transport: Error while dialing dial tcp [10.233.125.174:2379](http://10.233.125.174:2379/): connect: connection refused\""}
{"level":"warn","ts":"2022-12-01T06:36:06.823Z","logger":"etcd-client","caller":"v3@v3.5.4/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000d33500/upstream-pd:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest balancer error: last connection error: connection error: desc = \"transport: Error while dialing dial tcp [10.233.125.174:2379](http://10.233.125.174:2379/): connect: connection refused\""}
{"level":"info","ts":"2022-12-01T06:36:06.824Z","logger":"etcd-client","caller":"v3@v3.5.4/client.go:210","msg":"Auto sync endpoints failed.","error":"context deadline exceeded"}
{"level":"warn","ts":"2022-12-01T06:36:10.836Z","logger":"etcd-client","caller":"v3@v3.5.4/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000d33500/upstream-pd:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest balancer error: last connection error: connection error: desc = \"transport: Error while dialing dial tcp [10.233.125.174:2379](http://10.233.125.174:2379/): connect: connection refused\""}
panic: Table sink not found when updating resolved ts

goroutine 1032 [running]:
[go.uber.org/zap/zapcore.CheckWriteAction.OnWrite](http://go.uber.org/zap/zapcore.CheckWriteAction.OnWrite)(0x3?, 0x3?, {0x0?, 0x0?, 0xc006a1e4c0?})
        [go.uber.org/zap@v1.23.0/zapcore/entry.go:198](http://go.uber.org/zap@v1.23.0/zapcore/entry.go:198) +0x65
[go.uber.org/zap/zapcore.(*CheckedEntry).Write](http://go.uber.org/zap/zapcore.(*CheckedEntry).Write)(0xc005b67110, {0xc00622bc80, 0x3, 0x3})
        [go.uber.org/zap@v1.23.0/zapcore/entry.go:264](http://go.uber.org/zap@v1.23.0/zapcore/entry.go:264) +0x3ec
[go.uber.org/zap.(*Logger).Panic](http://go.uber.org/zap.(*Logger).Panic)(0xc00622bc80?, {0x38abbb6?, 0xc00427d9d8?}, {0xc00622bc80, 0x3, 0x3})
        [go.uber.org/zap@v1.23.0/logger.go:251](http://go.uber.org/zap@v1.23.0/logger.go:251) +0x59
[github.com/pingcap/log.Panic](http://github.com/pingcap/log.Panic)({0x38abbb6?, 0x31076c0?}, {0xc00622bc80?, 0x140?, 0x13f?})
        [github.com/pingcap/log@v1.1.1-0.20221116035753-734d527bc87c/global.go:54](http://github.com/pingcap/log@v1.1.1-0.20221116035753-734d527bc87c/global.go:54) +0x8b
[github.com/pingcap/tiflow/cdc/processor/sinkmanager.(*SinkManager).UpdateReceivedSorterResolvedTs](http://github.com/pingcap/tiflow/cdc/processor/sinkmanager.(*SinkManager).UpdateReceivedSorterResolvedTs)(0xc00053e3c0, 0x10fa, 0x613319b4bd80019)
        [github.com/pingcap/tiflow/cdc/processor/sinkmanager/manager.go:487](http://github.com/pingcap/tiflow/cdc/processor/sinkmanager/manager.go:487) +0x2ae
[github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/pebble.(*EventSorter).handleEvents](http://github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/pebble.(*EventSorter).handleEvents)(0xc002381600, 0xc002689b00, 0xc0023f6ae0)
        [github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/pebble/event_sorter.go:377](http://github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/pebble/event_sorter.go:377) +0xa9b
[github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/pebble.New.func1(0x7)](http://github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/pebble.New.func1(0x7))
        [github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/pebble/event_sorter.go:85](http://github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/pebble/event_sorter.go:85) +0xbb
created by [github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/pebble.New](http://github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/pebble.New)
        [github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/pebble/event_sorter.go:83](http://github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/pebble/event_sorter.go:83) +0x34a

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
/ # /tidb-server -V
Release Version: v6.5.0-alpha
Edition: Community
Git Commit Hash: f5487e3f2596cdd63ae9840606c7fff3fb539e03
Git Branch: heads/refs/tags/v6.5.0-alpha
```

Upstream TiKV version (execute `tikv-server --version`):

```console
/ # /tikv-server -V
TiKV
Release Version:   6.5.0-alpha
Edition:           Community
Git Commit Hash:   fbaaab32100292a54909b69649d15ee0e75fe58e
Git Commit Branch: heads/refs/tags/v6.5.0-alpha
```

TiCDC version (execute `cdc version`):

```console
bash-5.1# /cdc version
Release Version: v6.5.0-alpha
Git Commit Hash: 519c6f08dc5ca000d8c92101e2987daf071d716e
Git Branch: heads/refs/tags/v6.5.0-alpha
UTC Build Time: 2022-11-30 11:07:02
Go Version: go version go1.19.3 linux/amd64
Failpoint Build: false
```

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### What problem does this PR solve?

Issue Number: close #7762 .

### What is changed and how it works?
When close a processor, close sink before source.

#### Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/processor/processor.go [hunk 1]
```diff
@@ -188,7 +188,7 @@ func (p *processor) AddTable(
 				zap.Bool("isPrepare", isPrepare))
 			return true, nil
 		case tablepb.TableStateStopped:
-			log.Warn("The same table exists but is stopped. Cancel it and continue.",
+			log.Warn("The same table exists but is stopped, remove sink and redo records",
```
Comment (hicqu) [inline cdc/processor/processor.go:191]: revert

**Review by nongfushanquan — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

**Review by hi-rustin — APPROVED** [review]

**Review by amyangfei — APPROVED** [review]

**Comment by hicqu** [comment]
/run-integration-test tidb=f5487e3f2596cdd63ae9840606c7fff3fb539e03

**Comment by hicqu** [comment]
/run-integration-test tidb=f5487e3f2596cdd63ae9840606c7fff3fb539e03

**Comment by hicqu** [comment]
/run-integration-test tidb=f5487e3f2596cdd63ae9840606c7fff3fb539e03

**Pull request merged**